src/domain/app/routes/WithLocaleRoute.tsx

```diff
@@ -2,16 +2,19 @@ import React from 'react';
 
 import { useLocaleRouteNavigate } from './useLocaleRouteNavigate';
 
+const WithLocalRouteComponent = ({ Component, ...rest }: any) => {
+  useLocaleRouteNavigate();
+  return <Component {...rest} />;
+};
+
 /**
  * Check that the current pathname starts with a current locale.
  * If the current locale is not set in the pathname as the first segment,
  * the router will navigate to such a route and chagne the location pathname.
  */
 const WithLocalRoute = (WrappedComponent: React.FC) => {
   return ({ ...props }) => {
-    // eslint-disable-next-line react-hooks/rules-of-hooks
-    useLocaleRouteNavigate();
-    return <WrappedComponent {...props} />;
+    return <WithLocalRouteComponent Component={WrappedComponent} {...props} />;
   };
 };
```

Comment:
Reading the comment/documentation it seems `Locale` is meant here instead of `Local`, right?

Reply:
Right, it's a typo. Needs top be fixed.